### PR TITLE
[#161] Optimize equivalence predicate on outer/semi join

### PR DESCRIPTION
**Pull request by Seaven**
The PR purpose is optimize equivalence predicate on outer/semi join.

By the way, I think remove regular column check(#103 ) is unnecessary, so I added it back, and type of ConstantOperator is duplicate with ScalarOperator

### CODE REVIEW

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wangruin** [comment]
run all

**Comment by Seaven** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/test/java/com/starrocks/analysis/SelectStmtWithDecimalTypesNewPlannerTest.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public void testNullif() throws Exception {
         String thrift = UtFrameUtils.getPlanThriftStringForNewPlanner(ctx, sql);
         Assert.assertTrue(thrift.contains(expectString));
 
-        thrift = UtFrameUtils.getPlanThriftString(ctx, sql);
+        thrift = UtFrameUtils.getPlanThriftStringForNewPlanner(ctx, sql);
```
Comment (kangkaisen) [inline fe/fe-core/src/test/java/com/starrocks/analysis/SelectStmtWithDecimalTypesNewPlannerTest.java:68]: Why need to change it to new planner?

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/test/java/com/starrocks/analysis/SelectStmtWithDecimalTypesNewPlannerTest.java (hunk 1 above)
Comment (Seaven) [reply fe/fe-core/src/test/java/com/starrocks/analysis/SelectStmtWithDecimalTypesNewPlannerTest.java:68]: There is a problem with the be startup of FE-UT. By the way, migrate the old UT to new planner

**Comment by kangkaisen** [comment]
run starrocks_be_unittest

**Review by Youngwb — APPROVED** [review]

**Pull request merged**